internal/flagset/common_test.go

```diff
@@ -0,0 +1,129 @@
+	testCases := make([]FlagTest, 3)
+	testCases[0] = FlagTest{"sample", "KUBEFIRST_SAMPLE"}
+	testCases[1] = FlagTest{"sample-01", "KUBEFIRST_SAMPLE_01"}
+	testCases[2] = FlagTest{"sample-01-ab", "KUBEFIRST_SAMPLE_01_AB"}
+	for _, testCase := range testCases {
+		producedValue := GetFlagVarName(testCase.flag)
+		if producedValue != testCase.wanted {
+			t.Errorf("GetFlagVarName was incorrect, got: %s, want: %s.", producedValue, testCase.wanted)
+		}
+	}
+
+}
+
+// based on: https://gianarb.it/blog/golang-mockmania-cli-command-with-cobra
+
+func NewRootCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "hugo",
+		Short: "Hugo is a very fast static site generator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			flagValue, err := ReadConfigString(cmd, "sample")
+			if err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), err.Error())
+			}
+			fmt.Fprintf(cmd.OutOrStdout(), flagValue)
+			return nil
+		},
+	}
+	cmd.Flags().String("sample", "not-set", "This is a very important input.")
+	return cmd
+}
+
+func Test_DefineSource_set_by_flag(t *testing.T) {
+	cmd := NewRootCmd()
+	b := bytes.NewBufferString("")
+	os.Unsetenv("KUBEFIRST_SAMPLE")
+	cmd.SetOut(b)
+	cmd.SetArgs([]string{"--sample", "set-by-flag"})
+	cmd.Execute()
+	out, err := ioutil.ReadAll(b)
+	if err != nil {
+		t.Error(err)
+	}
+	if string(out) != "set-by-flag" {
+		t.Errorf("expected \"%s\" got \"%s\"", "set-by-flag", string(out))
+	}
+}
+
+func Test_DefineSource_set_by_var(t *testing.T) {
+	cmd := NewRootCmd()
+	b := bytes.NewBufferString("")
+	os.Setenv("KUBEFIRST_SAMPLE", "set-by-var")
+	cmd.SetOut(b)
+	cmd.Execute()
+	out, err := ioutil.ReadAll(b)
+	if err != nil {
+		t.Error(err)
+	}
+	if string(out) != "set-by-var" {
+		t.Errorf("expected \"%s\" got \"%s\"", "set-by-var", string(out))
+	}
+}
+
+func Test_DefineSource_notSet(t *testing.T) {
+	cmd := NewRootCmd()
+	b := bytes.NewBufferString("")
+	os.Unsetenv("KUBEFIRST_SAMPLE")
+	cmd.SetOut(b)
+	cmd.Execute()
+	out, err := ioutil.ReadAll(b)
+	if err != nil {
+		t.Error(err)
+	}
+	if string(out) != "not-set" {
+		t.Errorf("expected \"%s\" got \"%s\"", "not-set", string(out))
+	}
+}
+
+func NewRootCmdBool() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "hugo",
+		Short: "Hugo is a very fast static site generator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			flagValue, err := ReadConfigBool(cmd, "sample")
+			if err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), err.Error())
+			}
+			fmt.Fprintf(cmd.OutOrStdout(), strconv.FormatBool(flagValue))
+			return nil
+		},
+	}
+	cmd.Flags().Bool("sample", false, "This is a very important input.")
+	return cmd
+}
+
+func Test_DefineSource_set_by_flag_bool(t *testing.T) {
+	cmd := NewRootCmdBool()
+	b := bytes.NewBufferString("")
+	os.Setenv("KUBEFIRST_SAMPLE", "TRUE")
+	cmd.SetOut(b)
+	cmd.SetArgs([]string{"--sample"})
+	cmd.Execute()
+	out, err := ioutil.ReadAll(b)
+	if err != nil {
+		t.Error(err)
+	}
+	if string(out) != "true" {
```

Comment:
I guess you wanted `if string(out) != "TRUE" {` instead of "true"

Reply:
The code reflect what I was expecting, as if I do that: 
```
=== RUN   Test_DefineSource_set_by_flag_bool
2022/09/05 11:47:23 Flag(sample) set from CLI flag
    common_test.go:127: expected "true" got "true"
--- FAIL: Test_DefineSource_set_by_flag_bool (0.00s)
FAIL
FAIL	github.com/kubefirst/kubefirst/internal/flagset	0.007s
FAIL
```

If used: 

```golang
	out, err := ioutil.ReadAll(b)
	if err != nil {
		t.Error(err)
	}
	if string(out) != "TRUE" {
		t.Errorf("expected \"%s\" got \"%s\"", "true", string(out))
	}
```

I believe when golang translate bool to string, it is done lower case.

Reply:
https://stackoverflow.com/questions/38552803/how-to-convert-a-bool-to-a-string-in-go